src/v/cluster/cloud_metadata/offsets_snapshot.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "cluster/cloud_metadata/error_outcome.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "outcome.h"
+#include "serde/envelope.h"
+#include "utils/fragmented_vector.h"
+
+namespace cluster::cloud_metadata {
+
+struct group_offsets
+  : public serde::
+      envelope<group_offsets, serde::version<0>, serde::compat_version<0>> {
+    using rpc_adl_exempt = std::true_type;
+
+    struct partition_offset
+      : public serde::envelope<
+          partition_offset,
+          serde::version<0>,
+          serde::compat_version<0>> {
+        using rpc_adl_exempt = std::true_type;
+        partition_offset(model::partition_id p, kafka::offset o)
```

Comment:
i know that this wasn't introduced with this PR but just to start a discussion, maybe we should move `kafka::offset` out of the kafka module to `model::kafka_offset` this way we would avoid cluster depending on Kafka types. Actually maybe all this should live in Kafka module and cluster should only receive a metadata blob ?

Reply:
Yeah I think it seems reasonable. There's definitely some shame in polluting other namespaces with `kafka::`, and it'd definitely clarify certain pieces of code (thinking of the boundary between cloud/local storage and the kafka layer).

As for moving this snapshot over to `kafka`, I think maybe this is something longer term we can do, but for now I'd like to keep this where it is because it seems more flexible (e.g. recovery code will live in the `cluster::cloud_metadata` layer that will do some additional steps based on the deserialized struct, and it'd be nice to have that live outside the `kafka` layer)